### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by dmpetrov**
Add more tech details with less marketing keywords.

I expect more iterations on the readme. Let's merge this on asap and I'll create a 2nd version based on the feedback.

My feedback:
- reduce amount of text a bit
- show code examples sooner
- redesign GetStarted - more file/meta, less LLM

Please comment in Doc here: https://docs.google.com/document/d/1wPy_NSLOygwRAJSQxiBROQZahJHW1d-nYJFYUZBwl2w/edit

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -16,28 +16,87 @@
...
+🐍 **User-friendly interface.**
+   Operate efficiently with familiar Python objects and object fields, eliminating the
+   need for SQL.
 
 
 .. code:: console
 
    $ pip install datachain
 
-|Flowchart|
```
Comment (shcheklein) [inline README.rst:40]: is removal intentional? should we remove it from the bottom of the README then and from the repo (if it's not used in the docs anywhere else)

File: README.rst [hunk 2]
```diff
@@ -16,28 +16,87 @@
...
+
+Additional relatively new ideas:
+
+- **Functional style data processing.** Using a functional/chaining approach to data
+  processing rather than declarative SQL, inspired by R-dplyr and some Python libraries.
+- **Data Versioning.** Treats raw files in cloud storage as the source of truth for data
+  and implements data versioning, extending ideas from DVC (developed by the same team).
+
+
+What DataChain is NOT?
```
Comment (shcheklein) [inline README.rst:90]: I like this, any comparison helps a lot to understand a new tool better

File: README.rst [hunk 3]
```diff
@@ -16,28 +16,87 @@
 AI 🔗 DataChain
 ----------------
 
-DataChain is an open-source Python library for processing and curating unstructured
-data at scale.
+DataChain is a data-frame library designed for AI-specific scenarios. It helps ML and
```
Comment (shcheklein) [inline README.rst:19]: put out-of-memory?

File: README.rst [hunk 4]
```diff
@@ -16,28 +16,87 @@
...
+  - **Enrichments:** mapping, aggregating and generating using customer’s Python
+    code. This is needed to work with ML inference and LLM calls.
+
+- **Chain** is a sequence of operations on datasets. Chain executes operations in lazy
+  mode - only when needed.
+
+DataChain name comes from these major data structures: dataset and chaining.
+
+
+What’s new in DataChain?
```
Comment (shcheklein) [inline README.rst:67]: Title sounds a bit like a changelog ... probably should be "Why DataChain", or `What is unique in DataChain` ...

Comment (shcheklein) [inline README.rst:67]: it's a bit long tbh, and code is now pushed too much down I think

**Review by shcheklein — APPROVED** [review]

**Comment by dmpetrov** [comment]
I'm merging it - please feel free to keep feedback here or in the doc (preferred)

**Pull request merged**